js/bitget.js

```diff
@@ -3627,6 +3627,106 @@ module.exports = class bitget extends Exchange {
         };
     }
 
+    async fetchFundingHistory (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name bitget#fetchFundingHistory
+         * @description fetch the funding history
+         * @see https://bitgetlimited.github.io/apidoc/en/mix/#get-account-bill
+         * @param {string} symbol unified market symbol
+         * @param {int|undefined} since the starting timestamp in milliseconds
+         * @param {int|undefined} limit the number of entries to return
+         * @param {object} params extra parameters specific to the bitget api endpoint
+         * @returns {[object]} a list of [funding history structures]{@link https://docs.ccxt.com/en/latest/manual.html#funding-history-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (!market['swap']) {
+            throw new BadSymbol (this.id + ' fetchFundingHistory() supports swap contracts only');
+        }
+        if (since === undefined) {
+            since = this.milliseconds () - 31556952000; // 1 year
+        }
+        const request = {
+            'symbol': market['id'],
+            'marginCoin': market['quoteId'],
+            'startTime': since,
+            'endTime': this.milliseconds (),
+        };
+        if (limit !== undefined) {
+            request['pageSize'] = limit;
+        }
+        const response = await this.privateMixGetAccountAccountBill (this.extend (request, params));
+        //
+        //    {
+        //        "code": "00000",
+        //        "msg": "success",
+        //        "data": {
+        //            "result": [
+        //                {
+        //                    "id": "892962903462432768",
+        //                    "symbol": "ETHUSDT_UMCBL",
+        //                    "marginCoin": "USDT",
+        //                    "amount": "0",
+        //                    "fee": "-0.1765104",
+        //                    "feeByCoupon": "",
+        //                    "feeCoin": "USDT",
+        //                    "business": "contract_settle_fee",
+        //                    "cTime": "1648624867354"
+        //                }
+        //            ],
+        //            "endId": "885353495773458432",
+        //            "nextFlag": false,
+        //            "preFlag": false
+        //    }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const result = this.safeValue (data, 'result', []);
+        return this.parseFundingHistory (result, market, since, limit);
+    }
+
+    parseFundingHistory (contracts, market = undefined, since = undefined, limit = undefined) {
```

Comment:
@cheunhong We're not following the standard approach here, usually we create a method to handle multiple entries `parseFundingHistories(response, symbol, since, limit)`, then the parser to parse a single one `parseFundingHistory(info, market = undefined`) . Can you change it accordingly, pls?

Reply:
added